OnDeath.c

```diff
@@ -0,0 +1,144 @@
+bool OnDeath(int index = -1, int event = -1){
+	int p = yGetVarByIndex("allUnits", "player", index);
+	bool checkAgain = false;
+	switch(event)
+	{
+		case DEATH_DRAW_CARD:
+		{
+			drawCard(p);
+		}
+		case DEATH_OPPONENT_DRAW_CARD:
+		{
+			drawCard(3-p);
+		}
+		case DEATH_BOOM_SMALL:
+		{
+			trVectorQuestVarSet("pos", kbGetBlockPosition(""+1*yGetVarByIndex("allUnits", "tile", index)));
+			DeploySober("Meteor Impact Ground", "pos");
```

Comment:
deploySober uses teleports right? Teleports have a delay in multiplayer. You can use deployAtTile, which uses unit immediate garrison instead. Check board.c for deployAtTile